CONFIGURATION.md

```diff
@@ -0,0 +1,201 @@
+
+In order to make writing configurations easier and to enable re-use of configuration settings, we introduce a definitions container at the top level of the `helix-config.yaml` that can hold re-usable definitions like this:
+
+```yaml
+definitions:
+  strains:
+    base: &basestrain
+      code: /acapt/default/https---github-com-adobe-project-helix-io-git--master--
+      sticky: false
+    
+  origins:
+    publish: &publish
+      address: 192.168.0.1
+      name: publish
+      use_ssl: false
+```
+
+## Proposal: Configuration objects can be referenced using YAML references
+
+In order to re-use a complex configuration object (like an origin, a mount mapping, or even a strain), use YAML references (`*publish`) or YAML extensions (`<< *basestrain`). This allows us to rely on the behavior of the YAML parser instead of using home-grown inheritance and referencing logic that is likely less robust and certainly less well documented.
+
+A strain re-using the `*basestrain` above would look like this:
+
+```yaml
+  client:
+    <<: *basestrain
+    condition: req.http.host == "client.project-helix.io"
+    content:
+      repo: helix-cli
+      ref: master
+      owner: adobe
+    directoryIndex: readme.html
+```
+
+A strain referencing the `*publish` origin would look like this:
+```yaml
+  proxy-detailed:
+    sticky: true
+    origin: *publish
+```
+
+## Proposal: The `default` strain serves all traffic in production when no other strain is selected, therefore a `default` strain is mandatory
+
+We need one strain that is serving traffic that no other strain claims. This could be one by:
+
+1. requiring the existence of a `default` strain
+2. requiring the existence of one and only one strain that has the `default: true` property set
+
+Option (1) is easier to validate than option (2) and less error prone (every YAML editor will tell you if you have two keys called `default`), so I'd opt for (1)
+
+`hlx publish` should fail when no `default` strain exists.
+`hlx up` can continue without a `default` strain, but should issue a warning
+
+## Proposal: There is no implicit inheritance from `default`
+
+All inheritance becomes explicit, using the mechanism laid out in 
+https://github.com/adobe/helix-shared/issues/11#issuecomment-443122026. This means the `default` strain is special only in regards to serving traffic, but does not alter the behavior of any other strain.
+
+## Proposal: Strains can have a `mount` configuration that maps URL paths to content paths
+
+Right now, strains can have only one `content` node. With support for `mount`, a strain can have either a `content` or a `mount` property. Each `mount` object is a map between the URL path (key) and a `content` property (either git URL or object).
+
+```yaml
+strains:
+  default:
+    mount:
+      /: https://github.com/adobe/project-helix.io.git
+      /pipeline: https://github.com/adobe/helix-pipeline.git
+      /cli: https://github.com/adobe/helix-cli.git
+```
```

Comment:
why not have X strains for X mountpoints? otherwise you creating an implicit default for the `/` <-> `contentRepo` if there is no `mount` structure. with this approach, you cannot choose a different code repository for a mountpoint.

Reply:
We decided during the hackathon not to introduce mount points, so that it will be X strains for X mount points.

Reply:
Just so I'm clear here, this means each strain has at most a single mount point, is that correct?